handlers/commands/commands.js

```diff
@@ -22,27 +29,52 @@ const commandReference = `\
 <code>/user</code> - Shows user's status and warns.
 <code>/addcommand &lt;name&gt;</code> - to create a custom command.
 <code>/removecommand &lt;name&gt;</code> - to remove a custom command.
-
+`;
+const userCommands = `\
 <b>Commands for everyone</b>:
 <code>/staff</code> - Shows a list of admins.
 <code>/link</code> - Show the current group's link.
 <code>/groups</code> - Show a list of groups which the bot is admin in.
 <code>/report</code> - Reports the replied-to message to admins.
 `;
 
-const commandReferenceHandler = async ({ replyWithHTML }) => {
+const commandReferenceHandler = async (ctx) => {
 	const customCommands = await listCommands();
-	const customCommandsText = customCommands.length
-		? '\n<b>Custom commands:</b>\n' +
-		customCommands
-			.filter(command => command.isActive)
-			.sort((a, b) => a.role.toLowerCase() < b.role.toLowerCase())
+
+	const role = R.prop('role');
+	const customCommandsGrouped = R.groupBy(role, customCommands);
+	const userCustomCommands = '[everyone]\n<code>' +
+		customCommandsGrouped.everyone
 			.map(command =>
-				`[${command.role.toLowerCase()}] ` +
-				`<code>!${command.name}</code>`)
-			.join('\n')
+				`${command.name}`)
+			.join(', ') +
+			'</code>\n\n';
+
+	const adminCustomCommands = '[admins]\n<code>' +
+		customCommandsGrouped.admins
+			.map(command =>
+				`${command.name}`)
+			.join(', ') +
+			'</code>\n\n';
+
+	const masterCustomCommands = '[admin]\n<code>' +
+		customCommandsGrouped.master
+			.map(command =>
+				`${command.name}`)
+			.join(', ') +
+			'</code>\n\n';
```

Comment:
I'm actually fine with this duplication.

Unless you wanna display `[empty]` or `[no foo commands]` when the group is empty, I don't want that logic duplicated.

Reply:
I think just not showing the category should suffice.